# Heat Exchange Experimental Data Analysis
Presented by Rahil Shaik, Alex Quemel, Ethan Mibu, Sneha Kancharla

In [41]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Pre-processing

In [42]:
df_shell_cc = pd.read_csv("data/MW1_Shell_Counter_Current_formatted.csv")
df_shell_uni = pd.read_csv("data/MW1_Shell_Uni_Flow_formatted.csv")

df_tube_cross_350c_200h = pd.read_csv("data/tubular_cross_data/tubular_cross_350c_200h - tubular_counter_350c_300h.csv")
df_tube_cross_350c_300h = pd.read_csv("data/tubular_cross_data/tubular_counter_350c_300h - tubular_counter_350c_300h.csv")
df_tube_cross_350c_350h = pd.read_csv("data/tubular_cross_data/tubular_counter_350c_350h - tubular_counter_350c_350h.csv")
df_tube_cross_350c_400h = pd.read_csv("data/tubular_cross_data/tubular_counter_350c_400h - tubular_counter_350c_400h.csv")

df_tube_cross_350h_200c = pd.read_csv("data/tubular_cross_data/tubular_cross_350h_200c - tubular_counter_350c_200h.csv")
df_tube_cross_350h_300c = pd.read_csv("data/tubular_cross_data/tubular_cross_350h_300c - WL315C_2025-01-30_14-58-01.csv")
df_tube_cross_350h_350c = pd.read_csv("data/tubular_cross_data/tubular_cross_350h_350c - WL315C_2025-01-30_14-59-40.csv")
df_tube_cross_350h_400c = pd.read_csv("data/tubular_cross_data/tubular_cross_350h_400c - WL315C_2025-01-30_15-05-57.csv")

# df_tube_uni = pd.read_csv("data/MW1_Tube_Tube_Uni_Flow_formatted.csv")

df_plate_cc = pd.read_csv("data/plate_counter_extra.csv")
df_plate_uni = pd.read_csv("data/plate_parallel_extra.csv")

df_water_air_cc = pd.read_csv("data/finned_counter_extra.csv")
df_water_air_uni = pd.read_csv("data/finned_parallel_extra.csv")

In [43]:
df_tube_cross_350c_200h["hot flow (L/hr)"] = 200
df_tube_cross_350c_200h["cold flow (L/hr)"] = 350

df_tube_cross_350c_300h["hot flow (L/hr)"] = 300
df_tube_cross_350c_300h["cold flow (L/hr)"] = 350

df_tube_cross_350c_350h["hot flow (L/hr)"] = 350
df_tube_cross_350c_350h["cold flow (L/hr)"] = 350

df_tube_cross_350c_400h["hot flow (L/hr)"] = 400
df_tube_cross_350c_400h["cold flow (L/hr)"] = 350

df_tube_cross_350h_200c["hot flow (L/hr)"] = 350
df_tube_cross_350h_200c["cold flow (L/hr)"] = 200

df_tube_cross_350h_300c["hot flow (L/hr)"] = 350
df_tube_cross_350h_300c["cold flow (L/hr)"] = 300

df_tube_cross_350h_350c["hot flow (L/hr)"] = 350
df_tube_cross_350h_350c["cold flow (L/hr)"] = 350

df_tube_cross_350h_400c["hot flow (L/hr)"] = 350
df_tube_cross_350h_400c["cold flow (L/hr)"] = 400


In [44]:
df_tube_cross_350c_200h["hot flow (m^3/s)"] = df_tube_cross_350c_200h["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350c_200h["cold flow (m^3/s)"] = df_tube_cross_350c_200h["cold flow (L/hr)"] / (1000 * 3600)

df_tube_cross_350c_300h["hot flow (m^3/s)"] = df_tube_cross_350c_300h["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350c_300h["cold flow (m^3/s)"] = df_tube_cross_350c_300h["cold flow (L/hr)"] / (1000 * 3600)

df_tube_cross_350c_350h["hot flow (m^3/s)"] = df_tube_cross_350c_350h["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350c_350h["cold flow (m^3/s)"] = df_tube_cross_350c_350h["cold flow (L/hr)"] / (1000 * 3600)

df_tube_cross_350c_400h["hot flow (m^3/s)"] = df_tube_cross_350c_400h["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350c_400h["cold flow (m^3/s)"] = df_tube_cross_350c_400h["cold flow (L/hr)"] / (1000 * 3600)

df_tube_cross_350h_200c["hot flow (m^3/s)"] = df_tube_cross_350h_200c["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350h_200c["cold flow (m^3/s)"] = df_tube_cross_350h_200c["cold flow (L/hr)"] / (1000 * 3600)

df_tube_cross_350h_300c["hot flow (m^3/s)"] = df_tube_cross_350h_300c["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350h_300c["cold flow (m^3/s)"] = df_tube_cross_350h_300c["cold flow (L/hr)"] / (1000 * 3600)

df_tube_cross_350h_350c["hot flow (m^3/s)"] = df_tube_cross_350h_350c["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350h_350c["cold flow (m^3/s)"] = df_tube_cross_350h_350c["cold flow (L/hr)"] / (1000 * 3600)

df_tube_cross_350h_400c["hot flow (m^3/s)"] = df_tube_cross_350h_400c["hot flow (L/hr)"] / (1000 * 3600)
df_tube_cross_350h_400c["cold flow (m^3/s)"] = df_tube_cross_350h_400c["cold flow (L/hr)"] / (1000 * 3600)


In [45]:
dfs = [df_tube_cross_350c_200h, df_tube_cross_350c_300h, df_tube_cross_350c_350h, df_tube_cross_350c_400h,
       df_tube_cross_350h_200c, df_tube_cross_350h_300c, df_tube_cross_350h_350c, df_tube_cross_350h_400c ]

In [46]:
df_tube_cross_350c_200h.head(5)

Time  T1 (°C)  T2 (°C)  T3 (°C)  T4 (°C)  T5 (°C)  \
0  2/13/2025 4:29:46 PM.43     31.6      100     20.2     29.2     34.3   
1  2/13/2025 4:29:46 PM.93     31.6      100     20.2     29.2     34.3   
2  2/13/2025 4:29:47 PM.43     31.6      100     20.2     29.2     34.3   
3  2/13/2025 4:29:47 PM.93     31.6      100     20.2     29.2     34.3   
4  2/13/2025 4:29:48 PM.43     31.6      100     20.2     29.2     34.3   

   T6 (°C)  T7 (°C)  T8 (°C)  T9 (°C)  ...  dp1 (mbar)  dp2 (mbar)  \
0       22     36.5     24.5     39.2  ...        0.09           8   
1       22     36.5     24.5     39.2  ...        0.09           9   
2       22     36.5     24.6     39.2  ...        0.10           9   
3       22     36.5     24.5     39.2  ...        0.10           8   
4       22     36.5     24.5     39.2  ...        0.10           8   

   dQ1/dt (kW)  dQ2/dt (kW)  dQ3/dt (kW)  dQm/dt (kW)  hot flow (L/hr)  \
0       -23.72         3.64            0        13.68              200   
1       -23.85         3.63            0        13.74              200   
2       -23.81         3.66            0        13.74              200   
3       -23.96         3.65            0        13.81              200   
4       -23.95         3.63            0        13.79              200   

   cold flow (L/hr)  hot flow (m^3/s)  cold flow (m^3/s)  
0               350          0.000056           0.000097  
1               350          0.000056           0.000097  
2               350          0.000056           0.000097  
3               350          0.000056           0.000097  
4               350          0.000056           0.000097  

[5 rows x 21 columns]

## Analysis Question 1

## Analysis Question 3
For the tubular heat exchanger in counter-current flow configuration:

(a) Identify and report appropriate correlations for predicting the heat transfer
coefficients for the inner tube ($h_i$) and the annular region ($h_0$). Clearly state the
conditions and assumptions under which these correlations are valid (flow regime,
flow condition, characteristic length, etc.).

The first step in determining the appropriate correlations for the heat transfer coefficient in an inner tube is investigating the flow regime using Reynolds number, $Re$, to classify the flow as laminar or turbulent

$$
Re = \frac{\rho u D}{\mu} = \frac{4 \dot m}{\pi D \mu} 
$$

$$
\text{where} \  \dot m = \rho \dot V 
$$

We will use the LHS for classifying flow regime

In [47]:
mu = 0.001 # Pa * s
rho = 1000 # kg / m^3
D_inner = 6 * 10e-3 # m
D_annular = 13 * 10e-3 # m
D_hydraulic = D_annular - D_inner # m

A_i = np.pi * (D_inner ** 2) / 4
A_h = np.pi * ((D_annular ** 2) - (D_inner ** 2)) / 4

In [48]:
for df in dfs:
    u_cold = df["cold flow (m^3/s)"] / A_i
    u_hot = df["hot flow (m^3/s)"] / A_h
    df["Re_cold (inner tube)"] = (rho * u_cold * D_inner) / mu
    df["Re_hot (annular region)"] = (rho * u_hot * D_inner) / mu

In [49]:
df_tube_cross_350h_200c.head(5)

Time  T1 (°C)  T2 (°C)  T3 (°C)  T4 (°C)  T5 (°C)  \
0  2/13/2025 4:12:29 PM.43     27.7      100     17.9     25.8     30.7   
1  2/13/2025 4:12:29 PM.93     27.7      100     17.9     25.8     30.7   
2  2/13/2025 4:12:30 PM.43     27.7      100     17.8     25.8     30.7   
3  2/13/2025 4:12:30 PM.93     27.7      100     17.9     25.8     30.7   
4  2/13/2025 4:12:31 PM.43     27.6      100     17.8     25.8     30.7   

   T6 (°C)  T7 (°C)  T8 (°C)  T9 (°C)  ...  dQ1/dt (kW)  dQ2/dt (kW)  \
0     19.2     33.6     21.3     37.3  ...       -16.66         3.22   
1     19.2     33.6     21.3     37.3  ...       -16.66         3.20   
2     19.2     33.6     21.3     37.3  ...       -16.54         3.20   
3     19.2     33.6     21.3     37.3  ...       -16.42         3.18   
4     19.2     33.6     21.3     37.3  ...       -16.56         3.23   

   dQ3/dt (kW)  dQm/dt (kW)  hot flow (L/hr)  cold flow (L/hr)  \
0            0         9.94              350               200   
1            0         9.93              350               200   
2            0         9.87              350               200   
3            0         9.80              350               200   
4            0         9.90              350               200   

   hot flow (m^3/s)  cold flow (m^3/s)  Re_cold (inner tube)  \
0          0.000097           0.000056           1178.925504   
1          0.000097           0.000056           1178.925504   
2          0.000097           0.000056           1178.925504   
3          0.000097           0.000056           1178.925504   
4          0.000097           0.000056           1178.925504   

   Re_hot (annular region)  
0               558.438397  
1               558.438397  
2               558.438397  
3               558.438397  
4               558.438397  

[5 rows x 23 columns]

### Analysis of Heat Transfer Correlations for Tubular Heat Exchanger

In a double-pipe heat exchanger, **cold water** flows through the **inner tube**, and **hot water** flows through the **annular region**. The heat transfer coefficients $ h_i $ (inner tube) and $ h_o $ (annular region) are determined using empirical correlations that depend on the flow regime as indicated by the Reynolds number $ Re $.


### **Flow Regimes:**

| Regime        | Reynolds Number Condition |
|---------------|----------------------------|
| Laminar       | $ Re < 2300 $            |
| Transitional  | $ 2300 \leq Re \leq 4000 $ |
| Turbulent     | $ Re > 4000 $            |


### **Case-by-Case Correlations**

#### **Case 1: Inner Tube = Laminar, Annular Region = Laminar**

- **Assume constant wall temperature**
- Use:
$$
Nu = 3.66
\quad \Rightarrow \quad
h = \frac{Nu \cdot k}{D}
$$

#### **Case 2: Inner Tube = Laminar, Annular Region = Turbulent**

- **Inner Tube (Laminar):**
$$
Nu_i = 3.66
$$

- **Annular Region (Turbulent):**
$$
Nu_o = 0.023 \cdot Re^{0.8} \cdot Pr^{0.4}
$$

#### **Case 3: Inner Tube = Turbulent, Annular Region = Laminar**

- **Inner Tube (Turbulent):**
$$
Nu_i = 0.023 \cdot Re^{0.8} \cdot Pr^{0.3}
$$

- **Annular Region (Laminar):**
$$
Nu_o = 3.66
$$

#### **Case 4: Inner Tube = Turbulent, Annular Region = Turbulent**

- Both flows are turbulent:
$$
Nu_i = 0.023 \cdot Re^{0.8} \cdot Pr^{0.3}
\quad , \quad
Nu_o = 0.023 \cdot Re^{0.8} \cdot Pr^{0.4}
$$

#### **Transitional Flow in Either Region**

- **No reliable single correlation** exists for transitional flow.
- Best practice:
  - Interpolate between laminar and turbulent correlations
  - Or treat the region as "undefined regime" and flag for discussion

### **Summary of Correlations and Assumptions**

| Region         | Regime      | Correlation | Assumptions |
|----------------|-------------|-------------|-------------|
| Inner Tube     | Laminar     | $ Nu = 3.66 $ | Constant wall T, fully developed |
| Inner Tube     | Turbulent   | $ Nu = 0.023 Re^{0.8} Pr^{0.3} $ | Fully developed, smooth tube |
| Annular Region | Laminar     | $ Nu = 3.66 $ | Same as above |
| Annular Region | Turbulent   | $ Nu = 0.023 Re^{0.8} Pr^{0.4} $ | Fully developed, smooth annulus |

**Note:**  
- Characteristic length:  
  - Inner tube: $ D_i = 6\,\text{mm} $  
  - Annular region: $ D_h = D_o - D_i = 7\,\text{mm} $
- Valid for: $ 0.7 < Pr < 160 $


In [50]:
def classify_flow_regime(Re):
    if Re < 2300:
        return "Laminar"
    elif Re <= 4000:
        return "Transitional"
    else:
        return "Turbulent"

def classify_flow_in_dataframes(dfs):
    for df in dfs:
        df["Flow Regime (cold)"] = df["Re_cold (inner tube)"].apply(classify_flow_regime)
        df["Flow Regime (hot)"] = df["Re_hot (annular region)"].apply(classify_flow_regime)


In [51]:
classify_flow_in_dataframes(dfs)

In [55]:
df_tube_cross_350h_200c.head(5)

Time  T1 (°C)  T2 (°C)  T3 (°C)  T4 (°C)  T5 (°C)  \
0  2/13/2025 4:12:29 PM.43     27.7      100     17.9     25.8     30.7   
1  2/13/2025 4:12:29 PM.93     27.7      100     17.9     25.8     30.7   
2  2/13/2025 4:12:30 PM.43     27.7      100     17.8     25.8     30.7   
3  2/13/2025 4:12:30 PM.93     27.7      100     17.9     25.8     30.7   
4  2/13/2025 4:12:31 PM.43     27.6      100     17.8     25.8     30.7   

   T6 (°C)  T7 (°C)  T8 (°C)  T9 (°C)  ...  dQ3/dt (kW)  dQm/dt (kW)  \
0     19.2     33.6     21.3     37.3  ...            0         9.94   
1     19.2     33.6     21.3     37.3  ...            0         9.93   
2     19.2     33.6     21.3     37.3  ...            0         9.87   
3     19.2     33.6     21.3     37.3  ...            0         9.80   
4     19.2     33.6     21.3     37.3  ...            0         9.90   

   hot flow (L/hr)  cold flow (L/hr)  hot flow (m^3/s)  cold flow (m^3/s)  \
0              350               200          0.000097           0.000056   
1              350               200          0.000097           0.000056   
2              350               200          0.000097           0.000056   
3              350               200          0.000097           0.000056   
4              350               200          0.000097           0.000056   

   Re_cold (inner tube)  Re_hot (annular region)  Flow Regime (cold)  \
0           1178.925504               558.438397             Laminar   
1           1178.925504               558.438397             Laminar   
2           1178.925504               558.438397             Laminar   
3           1178.925504               558.438397             Laminar   
4           1178.925504               558.438397             Laminar   

   Flow Regime (hot)  
0            Laminar  
1            Laminar  
2            Laminar  
3            Laminar  
4            Laminar  

[5 rows x 25 columns]

(b) Using empirical correlations, calculate and report the inner tube ($h_i$) and annular region ($h_0$) heat transfer coefficients at a volumetric flow rate of 350 L/hr.

We will be using data from when both hot and cold streams are art 350 L/min given in `df_tube_cross_350h_350c`

In [57]:
df_tube_cross_350h_350c.head()


Time  T1 (°C)  T2 (°C)  T3 (°C)  T4 (°C)  T5 (°C)  \
0  1/30/2025 2:21:15 PM.19     32.9     48.0     20.8     30.1     35.2   
1  1/30/2025 2:21:15 PM.69     32.9     48.0     20.7     30.0     35.2   
2  1/30/2025 2:21:16 PM.19     32.9     48.0     20.8     30.0     35.2   
3  1/30/2025 2:21:16 PM.69     32.8     48.0     20.7     30.0     35.2   
4  1/30/2025 2:21:17 PM.19     32.8     48.1     20.8     30.1     35.1   

   T6 (°C)  T7 (°C)  T8 (°C)  T9 (°C)  ...  dQ3/dt (kW)  dQm/dt (kW)  \
0     22.5     37.3     25.0     39.6  ...            0         4.99   
1     22.5     37.3     25.0     39.6  ...            0         4.95   
2     22.5     37.3     25.0     39.7  ...            0         4.99   
3     22.5     37.3     25.0     39.6  ...            0         5.01   
4     22.5     37.3     25.0     39.6  ...            0         5.01   

   hot flow (L/hr)  cold flow (L/hr)  hot flow (m^3/s)  cold flow (m^3/s)  \
0              350               350          0.000097           0.000097   
1              350               350          0.000097           0.000097   
2              350               350          0.000097           0.000097   
3              350               350          0.000097           0.000097   
4              350               350          0.000097           0.000097   

   Re_cold (inner tube)  Re_hot (annular region)  Flow Regime (cold)  \
0           2063.119633               558.438397             Laminar   
1           2063.119633               558.438397             Laminar   
2           2063.119633               558.438397             Laminar   
3           2063.119633               558.438397             Laminar   
4           2063.119633               558.438397             Laminar   

   Flow Regime (hot)  
0            Laminar  
1            Laminar  
2            Laminar  
3            Laminar  
4            Laminar  

[5 rows x 25 columns]

Both flow regimes are laminar in this case, so we will use:

